roles/signals_test.py

```diff
@@ -0,0 +1,85 @@
+"""
+Tests for signals
+"""
+from django.db.models.signals import post_save
+from django.test import TestCase
+from factory.django import mute_signals
+from rolepermissions.verifications import (
+    has_role,
+    has_permission,
+    has_object_permission,
+)
+
+from courses.factories import ProgramFactory
+from roles.models import Role
+from profiles.factories import UserFactory
+
+
+# pylint: disable=no-self-use
+
+class SignalsTest(TestCase):
+    """
+    Tests for signals triggered by the role assignment
+    """
+
+    def setUp(self):
+        super(SignalsTest, self).setUp()
+        self.user = UserFactory.create()
+        self.program = ProgramFactory.create()
+
+    def test_assign_role(self):
+        """
+        Assigning the role with the model triggers a signal
+        to assign the correspondent role to the user.
+        """
+        assert has_role(self.user, 'staff') is False
+        assert has_permission(self.user, 'can_advance_search') is False
+        assert has_object_permission('can_advance_search', self.user, self.program) is False
+        Role.objects.create(
+            program=self.program,
+            user=self.user,
+            role='staff',
+        )
+        assert has_role(self.user, 'staff') is True
+        assert has_permission(self.user, 'can_advance_search') is True
+        assert has_object_permission('can_advance_search', self.user, self.program) is True
+
+    def test_presave_removes_current_role(self):
+        """
+        Updating the role in the model triggers a pre_save
+        signal that removes the previous role from the user.
+        """
+        mm_role = Role.objects.create(
+            program=self.program,
+            user=self.user,
+            role='staff',
+        )
+        assert has_role(self.user, 'staff') is True
+        assert has_permission(self.user, 'can_advance_search') is True
+        assert has_object_permission('can_advance_search', self.user, self.program) is True
+
+        with mute_signals(post_save):
+            mm_role.role = 'instructor'
```

Comment:
the actual Instructor role ([here](https://github.com/mitodl/micromasters/pull/709/files#diff-1ce2c7710978a1b0265fa651f240dde0R37)) is set to have the 'can_advance_search' = True by default. below, you are asserting that 'can_advance_search' is False. i don't think there is actually a Role associated with 'instructor' in the context of this test case, so it seems like you can set any value to `mm_role.role`. it's just confusing to set it to 'instructor' and expect 'can_advance_search' to be false when in the _actual_ instructor role, that should be true